src/jit/gschecks.cpp

```diff
@@ -440,8 +438,66 @@ void Compiler::gsParamsToShadows()
         gsShadowVarInfo[lclNum].shadowCopy = shadowVar;
     }
 
-    // Replace param uses with shadow copy
-    fgWalkAllTreesPre(gsReplaceShadowParams, (void*)this);
+    class ReplaceShadowParamsVisitor final : public GenTreeVisitor<ReplaceShadowParamsVisitor>
+    {
+        // Walk the locals of the method (i.e. GT_LCL_FLD and GT_LCL_VAR nodes) and replace the ones that correspond to
+        // "vulnerable" parameters with their shadow copies. If an original local variable has small type than replace
+        // the GT_LCL_VAR node type with TYP_INT.
+    public:
+        enum
+        {
+            DoPreOrder    = true,
+            DoLclVarsOnly = true
+        };
+
+        ReplaceShadowParamsVisitor(Compiler* compiler) : GenTreeVisitor<ReplaceShadowParamsVisitor>(compiler)
+        {
+        }
+
+        Compiler::fgWalkResult PreOrderVisit(GenTree** use, GenTree* user)
+        {
+            GenTree* tree = *use;
+            assert(tree != nullptr);
+            assert(tree->IsLocal());
+
+            unsigned int lclNum       = tree->AsLclVarCommon()->GetLclNum();
+            unsigned int shadowLclNum = m_compiler->gsShadowVarInfo[lclNum].shadowCopy;
+
+            if (shadowLclNum != NO_SHADOW_COPY)
+            {
+                LclVarDsc* varDsc = m_compiler->lvaTable + lclNum;
+                assert(ShadowParamVarInfo::mayNeedShadowCopy(varDsc));
+
+                tree->AsLclVarCommon()->SetLclNum(shadowLclNum);
+
+                if (varTypeIsSmall(varDsc->TypeGet()))
+                {
+                    if (user->OperIs(GT_ASG))
+                    {
+                        user->gtType = TYP_INT;
+                    }
+
+                    if (tree->OperIs(GT_LCL_VAR))
+                    {
+                        tree->gtType = TYP_INT;
+                    }
+                }
+            }
+
+            return WALK_CONTINUE;
+        }
+    };
+
+    BasicBlock* block;
+
+    foreach_block(this, block)
```

Comment:
I do not think such macros should be used. Most of the JIT code doesn't use it.

Reply:
I will remove